### PR TITLE
stmhal: Put all DMA channel & stream definitions in dma.h

### PR DESCRIPTION
**Pull request by dhylands**

### CODE REVIEW

File: stmhal/dma.h [hunk 1]
```diff
@@ -24,12 +24,59 @@
...
+// around each transfer.
+
+#define DMA_STREAM_I2C1_RX      DMA1_Stream0
+#define DMA_CHANNEL_I2C1_RX                     DMA_CHANNEL_1
+
+#define DMA_STREAM_I2C2_RX      DMA1_Stream2
+#define DMA_CHANNEL_I2C2_RX                     DMA_CHANNEL_7
+
+#define DMA_STREAM_I2C3_RX      DMA1_Stream2
+#define DMA_CHANNEL_I2C3_RX                     DMA_CHANNEL_3
```
Comment (dpgeorge) [inline stmhal/dma.h]: Within the same DMA stream, should the definitions be ordered on channel number as well?  This means you don't have to shuffle lines around if/when you reallocate the channel.

Comment (dhylands) [inline stmhal/dma.h]: That's probably not a bad idea (just for consistency). Although channels are more like alternate functions. I2C will only appear on one channel for a given stream.

I'll reorder and repush.

**Comment by dhylands** [comment]
Reordered to be by channel within stream.

**Comment by dpgeorge** [comment]
Rebased and merged in 0077958ad0d278c87620ee8ead9a9aea80f1663d.

**Pull request closed without merging**